clients/js/test/extensions/scaledUiAmountMint/amountToUiAmount.test.ts

```diff
@@ -0,0 +1,424 @@
+    epochStartTimestamp: 0n as UnixTimestamp,
+    leaderScheduleEpoch: 0n,
+    slot: 0n,
+    unixTimestamp: BigInt(unixTimestamp) as UnixTimestamp,
+  });
+  return populateMockAccount(data);
+}
+
+// GROUP 1: Basic functionality - standard mint without scaled UI extension
+test('should return the correct UiAmount when scaled ui amount config is not present', async (t) => {
+  const testCases = [
+    { decimals: 0, amount: BigInt(100), expected: '100' },
+    { decimals: 2, amount: BigInt(100), expected: '1' },
+    { decimals: 9, amount: BigInt(1000000000), expected: '1' },
+    { decimals: 10, amount: BigInt(1), expected: '1e-10' },
+    { decimals: 10, amount: BigInt(1000000000), expected: '0.1' },
+  ];
+
+  for (const { decimals, amount, expected } of testCases) {
+    const rpc = getMockRpc({
+      [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+      [mint]: createMockMintAccountInfo(decimals, false),
+    });
+    const result = await amountToUiAmountForMintWithoutSimulation(
+      rpc,
+      mint,
+      amount
+    );
+    t.is(result, expected);
+  }
+});
+
+// GROUP 2: Basic integer multiplier tests
+test('should return the correct UiAmount with scale factor of 2', async (t) => {
+  const testCases = [
+    { decimals: 0, amount: BigInt(100), expected: '200' },
+    { decimals: 1, amount: BigInt(100), expected: '20' },
+    { decimals: 10, amount: BigInt(10000000000), expected: '2' },
+  ];
+
+  for (const { decimals, amount, expected } of testCases) {
+    const rpc = getMockRpc({
+      [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+      [mint]: createMockMintAccountInfo(decimals, true, { multiplier: 2 }),
+    });
+
+    const result = await amountToUiAmountForMintWithoutSimulation(
+      rpc,
+      mint,
+      amount
+    );
+    t.is(result, expected);
+  }
+});
+
+test('should return the correct UiAmount with scale factor of 3', async (t) => {
+  const rpc = getMockRpc({
+    [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+    [mint]: createMockMintAccountInfo(10, true, { multiplier: 3 }),
+  });
+
+  const result = await amountToUiAmountForMintWithoutSimulation(
+    rpc,
+    mint,
+    BigInt(10000000000)
+  );
+  t.is(result, '3');
+});
+
+test('should return the correct amount with scale factor of 2', async (t) => {
+  const rpc = getMockRpc({
+    [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+    [mint]: createMockMintAccountInfo(0, true, { multiplier: 2 }),
+  });
+
+  const result = await uiAmountToAmountForMintWithoutSimulation(rpc, mint, '2');
+  t.is(result, 1n);
+});
+
+test('should return the correct amount with scale factor of 3', async (t) => {
+  const rpc = getMockRpc({
+    [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+    [mint]: createMockMintAccountInfo(10, true, { multiplier: 3 }),
+  });
+
+  const result = await uiAmountToAmountForMintWithoutSimulation(rpc, mint, '3');
+  t.is(result, 10000000000n);
+});
```

Comment:
nit: I liked how the other test cases were grouped, so can we have the same for these two?